src/adera/Machines/Rocket.cpp

```diff
@@ -163,9 +170,86 @@ void SysMachineRocket::update_physics()
     }
 }
 
+void SysMachineRocket::attach_plume_effect(ActiveEnt ent)
+{
+    using Magnum::GL::Mesh;
+    using Magnum::GL::Texture2D;
+    using Magnum::Trade::ImageData2D;
+    using namespace Magnum::Math::Literals;
+
+    ActiveEnt plumeNode = entt::null;
+
+    auto findPlumeHandle = [&](ActiveEnt ent)
+    {
+        ACompHierarchy& node = m_scene.reg_get<ACompHierarchy>(ent);
+        if (node.m_name.compare(0, 9, "fx_plume_") == 0)
+        {
+            plumeNode = ent;
+            return false;  // terminate search
+        }
+        return true;
+    };
+
+    m_scene.hierarchy_traverse(ent, findPlumeHandle);
+
+    if (plumeNode == entt::null)
+    {
+        std::cout << "ERROR: could not find plume anchor for MachineRocket "
+            << static_cast<int>(ent) << "\n";
+        return;
+    }
+    std::cout << "MachineRocket "
+        << static_cast<int>(ent) << "'s associated plume: "
```

Comment:
I think it would be better to define an operator<<(std::ostream & os, ActiveEnt val) { return os << static_cast<int>(val); }